src/app/excel-library/working-with/cells.component.ts

```diff
@@ -1,58 +1,244 @@
+import {
+    WorkbookExportService
+} from "../../utilities/excel-exporter";
+import { ExcelUtility } from "../../utilities/excel-utility";
 
 @Component({
-    changeDetection: ChangeDetectionStrategy.OnPush,
-    selector: "app-cells",
+    providers: [ WorkbookExportService ],
+    selector: "app-workbooks",
     styleUrls: ["./cells.component.scss"],
     templateUrl: "./cells.component.html"
 })
-export class ExcelLibraryWorkingWithCellsComponent implements AfterViewInit {
-
-    public data: any;
-
-    constructor() {
-
-        // TODO generate excel data or load data from .xls file
-        const usaMedals: any = [
-            { Year: "1996", UnitedStates: 148 },
-            { Year: "2000", UnitedStates: 142 },
-            { Year: "2004", UnitedStates: 134 },
-            { Year: "2008", UnitedStates: 131 },
-            { Year: "2012", UnitedStates: 135 },
-            { Year: "2016", UnitedStates: 146 }
-        ];
-        const chinaMedals: any = [
-            { Year: "1996", China: 110 },
-            { Year: "2000", China: 115 },
-            { Year: "2004", China: 121 },
-            { Year: "2008", China: 129 },
-            { Year: "2012", China: 115 },
-            { Year: "2016", China: 112 }
-        ];
-        const russiaMedals: any = [
-            { Year: "1996", Russia: 95 },
-            { Year: "2000", Russia: 91 },
-            { Year: "2004", Russia: 86 },
-            { Year: "2008", Russia: 65 },
-            { Year: "2012", Russia: 77 },
-            { Year: "2016", Russia: 88 }
-        ];
-        this.data = [ usaMedals, chinaMedals, russiaMedals ];
+export class ExcelLibraryWorkingWithCellsComponent implements OnInit {
+
+    @ViewChild("gridContainer", {read: ViewContainerRef})
+    public gridContainerRef: ViewContainerRef;
+    // openPlaceholder = "Choose Excel File:";
+    public canSave = false;
+    public wb: Workbook;
+    public ws: Worksheet;
+    public worksheetRegion: string[];
+    public selectedRegion: string;
+    public cellFeatures: string[];
+
+    constructor(private resolver: ComponentFactoryResolver, private injector: Injector) {
     }
 
     public ngOnInit() {
+        this.workbookCreate();
+    }
+
+    public workbookSave(): void {
+        if (this.canSave) {
+            // setting document properties to organize Excel files
+            this.wb.documentProperties.author = "My Name";
+            this.wb.documentProperties.company = "My Company";
+            this.wb.documentProperties.title = "Employees and income";
+            this.wb.documentProperties.status = "Completed";
+            this.wb.documentProperties.category = "Financial";
+            this.wb.documentProperties.keywords = "Financial;Company;Employees;income";
+
+            // setting protection on workbook of Excel file
+            this.wb.protection.allowEditStructure = true;
+            this.wb.protection.allowEditWindows = true;
+
+            this.wb.windowOptions.tabBarVisible = true;
+
```

Comment:
please remove settings for `documentProperties`, `protection`, `windowOptions `because these settings are features of workbooks and not cells.